beacon_node/lighthouse_network/gossipsub/src/types.rs

```diff
@@ -121,11 +123,16 @@ pub(crate) struct PeerConnections {
     pub(crate) sender: RpcSender,
     /// Subscribed topics.
     pub(crate) topics: BTreeSet<TopicHash>,
+    /// Don't send messages.
+    pub(crate) dont_send: LinkedHashMap<MessageId, Instant>,
```

Comment:
What do you think about using https://docs.rs/cached/latest/cached/stores/struct.TimedSizedCache.html?

Reply:
ah interesting, didn't know about it, thanks. But we already have the `hashlink` dependency in the repo and the flush code is just a couple of lines